kissnet.hpp

```diff
@@ -583,6 +598,12 @@ namespace kissnet
 				type	  = SOCK_STREAM;
 				iprotocol = IPPROTO_TCP;
 			}
+			
+			else if constexpr (sock_proto == protocol::tcp_ssl)
```

Comment:
It functionally doesn't change anything, but the code would be nicer without the duplication of the code between tcp and tcp_ssl.

I would propose to have one branch like this

```cpp
if constexpr(sock_proto == protocol::tcp || sock_proto == protocol::tcp_ssl)
```

Reply:
Done!